x/payments/handler.go

```diff
@@ -151,39 +154,39 @@ func handleMsgCreatePaymentTemplate(ctx sdk.Context, k Keeper, bk bank.Keeper, m
 		),
 	})
 
-	return sdk.Result{Events: ctx.EventManager().Events()}
+	return &sdk.Result{Events: ctx.EventManager().Events()}, nil
 }
 
 func handleMsgCreatePaymentContract(ctx sdk.Context, k Keeper, bk bank.Keeper,
-	msg MsgCreatePaymentContract) sdk.Result {
+	msg MsgCreatePaymentContract) (*sdk.Result, error) {
 
 	// Ensure that payment contract doesn't already exist
 	if k.PaymentContractExists(ctx, msg.PaymentContractId) {
-		return types.ErrAlreadyExists(types.DefaultCodespace, fmt.Sprintf(
-			"payment contract '%s' already exists", msg.PaymentContractId)).Result()
+		return nil, sdkerrors.Wrap(types.ErrAlreadyExists, fmt.Sprintf(
+			"payment contract '%s' already exists", msg.PaymentContractId))
```

Comment:
We could avoid using `fmt.Sprintf()` here by using `sdkerrors.Wrapf()`, like what is being done in other cases in this file.